src/inputs/labeled-inputs.scss

```diff
@@ -28,169 +45,165 @@
-    align-items: center;
-    font-size: $iui-font-size-small;
-    line-height: $iui-line-height;
-    margin-top: $iui-component-offset;
-    @include themed {
-      color: t(iui-text-color-muted);
-    }
+  &.iui-inline-label {
+    grid-template: 'label inputs icon' / auto 1fr auto;
+    grid-row-gap: 0;
 
-    svg {
-      width: $iui-icons-default;
-      height: $iui-icons-default;
-      display: flex; // needed for vertical alignment of webcomponents
-      margin-right: $iui-xs;
-      @include themed {
-        fill: t(iui-icons-color);
-      }
+    &.iui-with-message {
+      grid-template:
+        'label inputs inputs'
+        '. icon message' / auto auto 1fr;
     }
 
-    > .iui-borderless {
-      border-top-left-radius: 0;
-      border-bottom-left-radius: 0;
+    > .iui-input-group {
+      display: flex;
+      gap: $iui-m;
 
-      svg {
-        @include themed {
-          fill: t(iui-icons-color-actionable);
+      > .iui-checkbox,
+      > .iui-radio,
+      > .iui-toggle-switch {
+        margin-right: $iui-m;
+
+        @supports (gap: $iui-m) {
+          margin-right: 0;
         }
       }
-    }
 
-    a {
-      user-select: none;
+      &:not(:last-child) {
+        margin-right: $iui-m;
+      }
     }
-  }
 
-  // Appropriate spacing even if no text label is given
-  > .iui-checkbox,
-  > .iui-radio {
-    min-height: $iui-line-height;
-  }
-
-  &.iui-inline {
-    flex-direction: row;
-    align-items: center;
-
-    > .iui-label {
-      flex-shrink: 0;
-      margin: 0 $iui-m 0 0;
+    > .iui-input,
+    > .iui-textarea {
+      padding-right: $iui-icons-default + $iui-l;
 
-      &.iui-required {
-        margin-right: $iui-xs * 1.5;
+      &:last-child {
+        padding-right: $iui-sm;
       }
     }
+  }
+}
 
-    > .iui-checkbox:not(:last-child),
-    > .iui-radio:not(:last-child),
-    > .iui-toggle-switch:not(:last-child) {
-      margin-right: $iui-m;
-    }
-
-    > .iui-message {
-      margin: 0;
+/// Text label for an input.
+/// Supports .iui-required modifier to show red asterisk.
+@mixin iui-input-label {
+  font-weight: $iui-font-weight-semibold;
+  grid-area: label;
+  align-self: center;
 
-      svg:not(.iui-icon) {
-        margin-right: 0;
+  &.iui-required {
+    &::after {
+      content: ' *';
+      @include themed {
+        color: t(iui-color-foreground-negative);
       }
     }
+  }
+}
 
-    > .iui-input,
-    > .iui-textarea {
-      padding-right: $iui-icons-default + $iui-l;
+/// Modifier on iui-input-label to place it inline.
+@mixin iui-input-label-inline {
+  margin: 0 $iui-m 0 0;
 
-      + .iui-message {
-        position: absolute;
-        top: 50%;
-        right: 0;
-        transform: translateY(-50%);
+  &.iui-required {
+    margin-right: $iui-xs * 1.5;
+  }
+}
 
-        svg:not(.iui-icon) {
-          margin-right: $iui-sm;
-        }
-      }
+/// Message shown below input
+@mixin iui-input-message {
+  font-size: $iui-font-size-small;
```

Comment:
`line-height` needs to be changed.
![image](https://user-images.githubusercontent.com/81580355/132194861-847d842f-ccfd-4562-8c6e-63061de4ccaa.png)

Reply:
@FlyersPh9 I see we use same line-height for smaller font-size (12px). @bentleyvk said it looks better like this, but I tohught it should be a little less. As in the screenshot you can see, the spacing below input and above input looks different, because font size of message is smaller.